File: Python/pdf-endpoint.py
import requests

# Set the API endpoint URL
url = "https://api.pdfrest.com/pdf"

# Define the payload data for the request
payload = {
    'compression': 'lossy',
    'downsample': '300',
    'tagged_pdf': 'off',
    'output': 'pdfrest_pdf'
}

# Specify the file to be processed
files = [
    ('file', ('file', open('/path/to/file', 'rb'), 'application/octet-stream'))
]

# Set the headers, including the API key
headers = {
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'  # Replace with your API key
}

# Make a POST request to the API endpoint
response = requests.request(
    "POST", url, headers=headers, data=payload, files=files)

# Print the response content
print(response.text)

File: Node/pdf-endpoint.js
/**
The /pdf endpoint can take a single file, id, or url as input. 

This sample passes a jpeg file to the endpoint, but there's a variety of input file types that are accepted by this endpoint.
Import fetch
*/
import fetch, { FormData, fileFromSync } from "node-fetch";

// Append formdata here
let formdata = new FormData();
formdata.append("file", fileFromSync("../Sample_Input/rainbow.tif"));
formdata.append("output", "example_pdf_out");

let requestOptions = {
  method: "POST",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  },
  body: formdata,
  redirect: "follow",
};

// Define URL and submit request
fetch("https://api.pdfrest.com/pdf", requestOptions)
  .then((response) => response.text())
  .then((result) => console.log(result))
  .catch((error) => console.log("error", error));

// If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.js' sample.
